File: src/main/kotlin/app/web/database/HibernateUtil.java
package app.web.database;

import jakarta.persistence.EntityManager;
import jakarta.persistence.EntityManagerFactory;
import jakarta.persistence.Persistence;

public class HibernateUtil {
	
	private static final String PERSISTENCE_UNIT_NAME = "PointData";
	
	private static EntityManagerFactory entityManagerFactory;
	
	static {
		try {
			entityManagerFactory = Persistence.createEntityManagerFactory(PERSISTENCE_UNIT_NAME);
		} catch (Throwable ex) {
			System.err.println("Initial EntityManagerFactory creation failed." + ex);
			throw new ExceptionInInitializerError(ex);
		}
	}
	
	public static EntityManager getEntityManager() {
		return entityManagerFactory.createEntityManager();
	}
	
	public static void closeEntityManagerFactory() {
		if (entityManagerFactory != null && entityManagerFactory.isOpen()) {
			entityManagerFactory.close();
		}
	}
}

File: src/main/kotlin/app/web/database/HibernateUtil.kt
package app.web.database

import jakarta.persistence.EntityManager
import jakarta.persistence.EntityManagerFactory
import jakarta.persistence.Persistence

class HibernateUtil() {
    private var entityManagerFactory : EntityManagerFactory = Persistence.createEntityManagerFactory("PointData")
    fun getEntityManager() : EntityManager {
        return entityManagerFactory.createEntityManager()
    }

    fun closeEntityManagerFactory() {
        if (entityManagerFactory.isOpen) {
            entityManagerFactory.close()
        }
    }
}